### PR TITLE
Implement cache from

### PR DESCRIPTION
**Pull request by thomaschaaf**
Fixes: https://github.com/docker/compose/issues/4309
Depends on: https://github.com/docker/docker-py/pull/1383

Signed-off-by: Thomas Schaaf <schaaf@komola.de>

### CODE REVIEW

File: tests/integration/service_test.py [hunk 1]
```diff
@@ -594,6 +594,20 @@ def test_build_with_build_args(self):
         service.build()
         assert service.image()
 
+    @v2_only()
```
Comment (thomaschaaf) [inline tests/integration/service_test.py]: Is this correct?

File: compose/config/config_schema_v2.0.json [hunk 2]
```diff
@@ -56,9 +56,10 @@
             {
               "type": "object",
               "properties": {
+                "args": {"$ref": "#/definitions/list_or_dict"},
+                "cachefrom": {"$ref": "#/definitions/list_of_strings"},
```
Comment (thomaschaaf) [inline compose/config/config_schema_v2.0.json]: Am I allowed to add it to the 2.0 schema?

**Comment by jyapayne** [comment]
Since the [docker-py issue](https://github.com/docker/docker-py/pull/1417) was merged in, this can be retested and reviewed, no?

**Comment by shin-** [comment]
Closing via #4514 - Thanks!

**Pull request closed without merging**